tests/modify_permission.rs

```diff
@@ -0,0 +1,276 @@
+        false,
+        "Test User",
+        PermissionEnum::EditSeqSec,
+        Some("test_seq,1".to_string()));
+
+    // Make sure changes were saved
+    common::assert_repo_no_modified_files(&root.path());
+
+}
+
+#[test]
+#[should_panic(expected = "Invalid permission target")]
+fn fails_with_bad_target_editseq() {
+    let root = setup::setup_init_cd();
+    let root_private_key_path = common::make_key_file(root.path(), "root.pem", TestKey::RootKeyPem);
+
+    // Create user
+    setup::try_new_user(
+        root_private_key_path.as_path(),
+        root.path(),
+        "Test User",
+        "a.pub",
+        TestKey::GoodKeyPub);
+
+    // Create sequence
+    setup::try_make_sequence(&root_private_key_path, "test_seq", "Dissonance.ogg");
+
+    // Try to add permission to user
+    try_set_permission(
+        &root_private_key_path,
+        true,
+        "Test User",
+        PermissionEnum::EditSeq,
+        Some("nonexistent".to_string()));
+
+}
+
+#[test]
+#[should_panic(expected = "Invalid permission target")]
+fn fails_with_bad_target_editseqsec() {
+    let root = setup::setup_init_cd();
+    let root_private_key_path = common::make_key_file(root.path(), "root.pem", TestKey::RootKeyPem);
+
+    // Create user
+    setup::try_new_user(
+        root_private_key_path.as_path(),
+        root.path(),
+        "Test User",
+        "a.pub",
+        TestKey::GoodKeyPub);
+
+    // Create sequence
+    setup::try_make_sequence(&root_private_key_path, "test_seq", "Dissonance.ogg");
+
+    // Try to add permission to user
+    try_set_permission(
+        &root_private_key_path,
+        true,
+        "Test User",
+        PermissionEnum::EditSeqSec,
+        Some("section999".to_string()));
+}
+
+#[test]
+#[should_panic(expected = "entity not found")]
+fn fails_with_bad_path_to_private_key() {
+    let root = setup::setup_init_cd();
+    let root_private_key_path = Path::new("undefined.pem");
+
+    setup::try_new_user(
+        root_private_key_path,
+        root.path(),
+        "Test User",
+        "a.pub",
+        TestKey::GoodKeyPub);
+
+    try_set_permission(&root_private_key_path, true, "Test User", PermissionEnum::Administrate, None);
+}
+
+#[test]
+fn works_trading_admin_power() {
+    let root = setup::setup_init_cd();
+    let root_private_key_path = common::make_key_file(root.path(), "root.pem", TestKey::RootKeyPem);
+    let admin2_private_key_path = common::make_key_file(root.path(), "b.pem", TestKey::GoodKeyPem);
+
+    // Setup new user with Administrate permission
+    setup::try_new_user(
+        &root_private_key_path.as_path(),
+        root.path(),
+        "Admin2",
+        "b.pub",
+        TestKey::GoodKeyPub);
+    try_set_permission(&root_private_key_path, true, "Admin2", PermissionEnum::Administrate, None);
+
+    // Now have that new user take away the first's Administrate permission
+    try_set_permission(&admin2_private_key_path, false, "root", PermissionEnum::Administrate, None);
```

Comment:
Wait actually, while I approved of an Admin being able to take away a different admin's privilege, I don't think you should be able to deprivilege root. Root is like, root. It's the backup.